gdx/test/com/badlogic/gdx/math/FloatCounterTest.java

```diff
@@ -0,0 +1,152 @@
+		counter2.reset();
+		Assert.assertNull(counter2.mean);
+		Assert.assertEquals(0, counter2.count);
+		Assert.assertEquals(0.0f, counter2.total, 0);
+		Assert.assertEquals(Float.MAX_VALUE, counter2.min, 0);
+		Assert.assertEquals(-Float.MAX_VALUE, counter2.max, 0);
+		Assert.assertEquals(0.0f, counter2.average, 0);
+		Assert.assertEquals(0.0f, counter2.latest, 0);
+		Assert.assertEquals(0.0f, counter2.value, 0);
+		counter2.put(1.0f);
+		Assert.assertEquals(1.0f, counter2.latest, 0);
+		Assert.assertEquals(1.0f, counter2.total, 0);
+		Assert.assertEquals(1.0f, counter2.average, 0);
+		Assert.assertEquals(1.0f, counter2.value, 0);
+		Assert.assertEquals(1, counter2.count);
+		Assert.assertEquals(1.0f, counter2.max, 0);
+		Assert.assertEquals(1.0f, counter2.min, 0);
+		counter2.put(34.0f);
+		counter2.put(15.24f);
+		counter2.put(43.62f);
+		Assert.assertEquals(43.62f, counter2.latest, 0);
+		Assert.assertEquals(93.86f, counter2.total, 0);
+		Assert.assertEquals(23.465f, counter2.average, 0);
+		Assert.assertEquals(43.62f, counter2.value, 0);
+		Assert.assertEquals(4, counter2.count);
+		Assert.assertEquals(43.62f, counter2.max, 0);
+		Assert.assertEquals(1.0f, counter2.min, 0);
+	}
+
+	@Test
+	public void testLimitCases () {
+		// Tests with a Windowed mean
+		counter1.reset();
+		counter1.put(Float.MIN_NORMAL);
+		Assert.assertEquals(1.17549435E-38f, counter1.latest, 0);
+		counter1.put(10000.85f);
+		Assert.assertEquals(10000.850000000000000000000000000000000000117549435f, counter1.total, 0);
+		counter1.put(75.15f);
+		counter1.put(7E18f);
+		Assert.assertEquals(1750000000000002519.00000000000000000000000000000000000002938735875f, counter1.average, 0);
+		counter1.put(5f);
+		Assert.assertEquals(1400000000000002013.80000000000000000000000000000000000002350988700f, counter1.mean.getMean(), 0);
+
+		// Tests without a Windowed mean
+		counter2.reset();
+		try {
+			counter2.put(Float.NaN);
+			Assert.fail();
+		} catch (RuntimeException e) {
+			// This should throw an exception as a FloatCounter cannot handle a NaN
+		}
+		counter2.put(0.000000000000001f);
+		counter2.put(1e14f);
+		Assert.assertEquals(100000000000000.000000000000001f, counter2.total, 0);
+		Assert.assertEquals(50000000000000.0000000000000005f, counter2.average, 0);
+		counter2.put(Float.MAX_VALUE);
+		Assert.assertEquals(Float.MAX_VALUE, counter2.total, 0);
+		counter2.put(-Float.MAX_VALUE);
+		/*
+		 * Because we put the max value and the total cannot be more than max value (except infinity), the new total will be 0
+		 * (should we prohibit MAX_VALUE and INFINITY to avoid that?)
+		 */
+		Assert.assertEquals(0f, counter2.total, 0);
+		counter2.put(Float.POSITIVE_INFINITY);
+		Assert.assertEquals(Float.POSITIVE_INFINITY, counter2.total, 0);
+		counter2.put(Float.NEGATIVE_INFINITY);
+		/*
+		 * After implementing this next test, I realized that putting POSITIVE_INFINITY then NEGATIVE_INFINITY, the total was not a
+		 * number (same as before, should this be prohibited ?)
```

Comment:
Consider marking your pull request as draft if it contains questions, since this doesn't look ready to be merged.

Reply:
also done thank you